spl/src/token_2022_extensions/token_metadata.rs

```diff
@@ -0,0 +1,118 @@
+use anchor_lang::solana_program::account_info::AccountInfo;
+use anchor_lang::solana_program::pubkey::Pubkey;
+use anchor_lang::Result;
+use anchor_lang::{context::CpiContext, Accounts};
+
+use borsh::{BorshDeserialize, BorshSerialize};
+use spl_pod::optional_keys::OptionalNonZeroPubkey;
+use spl_token_metadata_interface::state::Field;
+
+#[derive(Clone, BorshDeserialize, BorshSerialize)]
+pub struct TokenMetadataInitializeArgs {
+    pub name: String,
+    pub symbol: String,
+    pub uri: String,
+}
+
+pub fn token_metadata_initialize<'info>(
+    ctx: CpiContext<'_, '_, '_, 'info, TokenMetadataInitialize<'info>>,
+    args: TokenMetadataInitializeArgs,
```

Comment:
Any specific reason why this function takes a struct argument? Ideally we don't need to use `borsh` here.

Reply:
No reason, will update to use direct args